File: Section 20 (Object-C Actions)/Temperature Calculator/Temperature Calculator/ViewController.m
//
//  ViewController.m
//  Temperature Calculator
//
//  Created by InfoSapex Limited on 1/11/19.
//  Copyright © 2019 Muiz Ahmed Khan. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
}


- (IBAction)enterTextinLabel:(id)sender {
    if(self.calculateSegmentControl.selectedSegmentIndex == 0){
        double far = [self.textfield.text doubleValue];
        double cel = (far - 32)/1.8;
        
        self.outputLabel.text = [NSString stringWithFormat:@"%.2f Celsius", cel];
        
        if(cel > 120){
            self.image.image = [UIImage imageNamed:@"Temp9.png"];
        }else if(cel > 100){
            self.image.image = [UIImage imageNamed:@"Temp8.png"];
        }else if(cel > 80){
            self.image.image = [UIImage imageNamed:@"Temp7.png"];
        }else if(cel > 60){
            self.image.image = [UIImage imageNamed:@"Temp6.png"];
        }else if(cel > 40){
            self.image.image = [UIImage imageNamed:@"Temp5.png"];
        }else if(cel > 20){
            self.image.image = [UIImage imageNamed:@"Temp4.png"];
        }else if(cel > 0){
            self.image.image = [UIImage imageNamed:@"Temp3.png"];
        }else if(cel > -20){
            self.image.image = [UIImage imageNamed:@"Temp2.png"];
        }else if(cel < -20){
            self.image.image = [UIImage imageNamed:@"Temp1.png"];
        }
    }else{
        double cel = [self.textfield.text doubleValue];
        double far = (cel*1.8)+32;
        
        self.outputLabel.text = [NSString stringWithFormat:@"%.2f", far];
        
        if(far > 180){
            self.image.image = [UIImage imageNamed:@"Temp9.png"];
        }else if(far > 160){
            self.image.image = [UIImage imageNamed:@"Temp8.png"];
        }else if(far > 140){
            self.image.image = [UIImage imageNamed:@"Temp7.png"];
        }else if(far > 120){
            self.image.image = [UIImage imageNamed:@"Temp6.png"];
        }else if(far > 100){
            self.image.image = [UIImage imageNamed:@"Temp5.png"];
        }else if(far > 80){
            self.image.image = [UIImage imageNamed:@"Temp4.png"];
        }else if(far > 40){
            self.image.image = [UIImage imageNamed:@"Temp3.png"];
        }else if(far > 20){
            self.image.image = [UIImage imageNamed:@"Temp2.png"];
        }else if(far < 20){
            self.image.image = [UIImage imageNamed:@"Temp1.png"];
        }
    }
    
}
- (IBAction)enterTextLabel:(id)sender {
    
    
}

- (IBAction)convertion:(id)sender {
    if(self.calculateSegmentControl.selectedSegmentIndex == 0){
        self.enterLabel.text = @"Enter Fahrenheit";
        self.textfield.text = @"";
                   
        self.outputLabel.text = [NSString stringWithFormat:@"0 Celsius"];
    }else{
        self.enterLabel.text = @"Enter Celsius";
        self.textfield.text = @"";
                   
        self.outputLabel.text = [NSString stringWithFormat:@"0 Fahrenheit"];
    }
    
}
@end

File: Section 4 (Swift 4 Functions)/Fancy Text/Fancy Text/ViewController.swift
//
//  ViewController.swift
//  Fancy Text
//
//  Created by InfoSapex Limited on 9/10/19.
//  Copyright © 2019 Muiz Ahmed Khan. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    
    @IBOutlet weak var label: UILabel!
    @IBOutlet weak var textField: UITextField!
    @IBOutlet weak var shadow: UIButton!
    
    var fontsize: CGFloat = 30
    var state = false
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
    }
    
    
    @IBAction func enterText(_ sender: Any) {
        label.text = textField.text
        self.resignFirstResponder()
    }
    
    @IBAction func redBtn(_ sender: Any) {
        label.textColor = UIColor.red
    }
    
    @IBAction func greenBtn(_ sender: Any) {
        label.textColor = UIColor.green
    }
    
    @IBAction func blueBtn(_ sender: Any) {
        label.textColor = UIColor.blue
    }
    
    @IBAction func font_1Btn(_ sender: Any) {
        label.font = UIFont(name: "SugarstyleMillenial-Regular" , size:30)
    }
    
    @IBAction func font_2Btn(_ sender: Any) {
        label.font = UIFont(name: "MoonFlower" , size:fontsize)
    }
    
    @IBAction func font_3Btn(_ sender: Any) {
        label.font = UIFont(name: "LemonMilk" , size:fontsize)
    }
    
    @IBAction func font_4Btn(_ sender: Any) {
         label.font = UIFont(name: "Blacksword" , size:fontsize)
    }
    
    @IBAction func shadowBtn(_ sender: Any) {
        if(state == false){
            label.layer.shadowColor = UIColor.black.cgColor
            label.layer.shadowOffset = CGSize(width:2, height: 2)
            label.layer.shadowRadius = 2
            label.layer.shadowOpacity = 0.25
            
            state = true
        }else{
            label.layer.shadowOpacity = 0
            
            state = false
        }
    }
    
    @IBAction func smallBtn(_ sender: Any) {
        fontsize = 30
        label.font = label.font.withSize(fontsize)
    }
    
    @IBAction func mediumBtn(_ sender: Any) {
         fontsize = 50
         label.font = label.font.withSize(fontsize)
    }
    
    @IBAction func largeBtn(_ sender: Any) {
         fontsize = 80
         label.font = label.font.withSize(fontsize)
    }
    
}

